File: lecture007/wave2.cpp
#include<iostream>
#include<vector>
using namespace std;
void wave1(vector<vector<int>> &arr){
    int m,n;
    n=arr.size();
    m=arr[0].size();
    for(int col=0; col<n ; col++){
        if(col%2==0){
            for(int row=0 ; row<m ; row++){
                cout<<arr[row][col]<<" ";
            }
        }
        else{
            for(int row=m-1 ; row>=0 ; row--){
                cout<<arr[row][col]<<" ";
            }
        }
    }
}
void input(vector<vector<int>> &arr){
    for(int col=0 ; col<arr.size() ; col++){
        for(int row=0 ; row< arr[0].size() ; row++){
            cin>>arr[col][row];
        }
    }
}
void display(vector<vector<int>> &arr){
    for(int col=0 ; col<arr.size() ; col++){
        for(int row=0 ; row< arr[0].size() ; row++){
            cout<<arr[col][row]<<" ";
        }
    }
}

int main(int args , char**argv){
    int row=0 , col=0;
    cin>>row>>col;
    vector<vector<int>> arr(row , vector<int>(col,0));
    input(arr);
    cout<<endl;
    wave1(arr);
    //display(arr);
}

File: lecture007/wave2.java
import java.util.Scanner;
public class wave2{
    public static Scanner scn = new Scanner(System.in);
    public static void main(String[] args){
        System.out.println("Enter row and col values : ");
        int row=scn.nextInt();
        int col=scn.nextInt();
        int [][] arr=new int[row][col];
        System.out.println("Enter array : ");
        input(arr);
        System.out.println("new array is ");
        wave(arr);
    }

    public static void input(int [][] arr ){
        for(int row=0 ; row<arr.length ; row++){
            for(int col=0 ; col<arr[0].length ; col++){
                arr[row][col]=scn.nextInt();
            }
        }
    }

    public static void wave(int [][] arr){
        int m=arr.length;
        int n=arr[0].length;
        for(int col=0 ; col<m ; col++){
            if(col%2==0){
                for(int row=0 ; row<n ; row++)
                System.out.print(arr[row][col] + " ");
            }
            else{
                for(int row=n-1 ; row>=0 ; row--)
                System.out.print(arr[row][col] + " ");
            }
        }

    }

}